core/resources/polygon.vs

```diff
@@ -1,30 +1,30 @@
-#ifdef GL_ES
-precision mediump float;
-#endif
-
+uniform mat4 u_modelView;
 uniform mat4 u_modelViewProj;
-uniform vec3 u_lightDirection;
+uniform float u_time;
+
+#pragma tangram: material
+#pragma tangram: vert_lighting
 
 attribute vec4 a_position;
 attribute vec4 a_color;
 attribute vec3 a_normal;
 attribute vec2 a_texcoord;
 
-varying vec4 v_pos;
 varying vec4 v_color;
+varying vec3 v_eyeToPoint;
 varying vec3 v_normal;
 varying vec2 v_texcoord;
 
 void main() {
 
-	v_normal = a_normal;
+  	v_normal = normalize(a_normal);
+  	v_texcoord = a_texcoord;
 
-  	v_color = a_color;
-  	
-  	float lit = dot(normalize(u_lightDirection), normalize(a_normal));
-  	v_color.rgb *= clamp(lit * 1.5, 0.5, 1.5);
-  	
-	v_texcoord = a_texcoord;
+	v_eyeToPoint = vec3(u_modelView * a_position);
+	calculateLighting(v_eyeToPoint,v_normal);
```

Comment:
So, the `calculateLighting` function gets injected into both vertex and fragment shaders, but can do different pieces of work in each, e.g. directional light can be calculated in the vertex shader, then additional point light computations can be added in the fragment shader call of the same function name (but different function body), with the intermediate ? Is that correct? We'll also want to document this more clearly.